### PR TITLE
feat: add popupProps to Callout

### PR DESCRIPTION
**Pull request by smhigley**
Fixes [14807](https://dev.azure.com/microsoftdesign/fluent-ui/_workitems/edit/14807)

Adding a `popupProps` prop to Callout allows authors to pass through props directly to the underlying `Popup` element, which allows for e.g. making a modal Callout (the original request).

Panel also uses `popupProps` and Callout already has `layerProps`, so this should be consistent with existing patterns.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/f1bb447bb76b4bbc15cb9c377f7543cd06f83887'>f1bb447bb76b4bbc15cb9c377f7543cd06f83887</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/BaseButton-mount_base.html">1068</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/BaseButton-mount.html">1061</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Breadcrumb-mount_base.html">2940</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Breadcrumb-mount.html">2968</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Checkbox-mount_base.html">2941</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Checkbox-mount.html">2911</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/CheckboxBase-mount_base.html">2546</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/CheckboxBase-mount.html">2601</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ChoiceGroup-mount_base.html">5403</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ChoiceGroup-mount.html">5422</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ComboBox-mount_base.html">1125</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ComboBox-mount.html">1079</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/CommandBar-mount_base.html">11529</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/CommandBar-mount.html">11488</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ContextualMenu-mount_base.html">16424</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ContextualMenu-mount.html">12336</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DefaultButton-mount_base.html">1362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DefaultButton-mount.html">1361</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DetailsRow-mount_base.html">4409</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DetailsRow-mount.html">4313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DetailsRowFast-mount_base.html">4273</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DetailsRowFast-mount.html">4388</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DetailsRowNoStyles-mount_base.html">4163</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DetailsRowNoStyles-mount.html">4158</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Dialog-mount_base.html">3154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Dialog-mount.html">3119</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DocumentCardTitle-mount_base.html">212</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/DocumentCardTitle-mount.html">204</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Dropdown-mount_base.html">3829</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Dropdown-mount.html">3862</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/FocusTrapZone-mount_base.html">2091</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/FocusTrapZone-mount.html">2163</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/FocusZone-mount_base.html">1992</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/FocusZone-mount.html">2006</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/IconButton-mount_base.html">2051</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/IconButton-mount.html">2084</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Label-mount_base.html">384</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Label-mount.html">394</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Layer-mount_base.html">3601</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Layer-mount.html">3579</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Link-mount_base.html">551</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Link-mount.html">560</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/MenuButton-mount_base.html">1774</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/MenuButton-mount.html">1743</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/MessageBar-mount_base.html">2564</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/MessageBar-mount.html">2527</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Nav-mount_base.html">3999</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Nav-mount.html">3917</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/OverflowSet-mount_base.html">1287</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/OverflowSet-mount.html">1173</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Panel-mount_base.html">2457</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Panel-mount.html">2486</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Persona-mount_base.html">1160</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Persona-mount.html">1198</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Pivot-mount_base.html">1647</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Pivot-mount.html">1663</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/PrimaryButton-mount_base.html">1519</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/PrimaryButton-mount.html">1538</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Rating-mount_base.html">8993</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Rating-mount.html">8941</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SearchBox-mount_base.html">1550</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SearchBox-mount.html">1527</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Shimmer-mount_base.html">2851</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Shimmer-mount.html">2891</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Slider-mount_base.html">2216</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Slider-mount.html">2209</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SpinButton-mount_base.html">5762</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SpinButton-mount.html">5689</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Spinner-mount_base.html">512</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Spinner-mount.html">508</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SplitButton-mount_base.html">3610</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SplitButton-mount.html">3766</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Stack-mount_base.html">620</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Stack-mount.html">640</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/StackWithIntrinsicChildren-mount_base.html">2794</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/StackWithIntrinsicChildren-mount.html">2711</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/StackWithTextChildren-mount_base.html">6373</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/StackWithTextChildren-mount.html">6359</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SwatchColorPicker-mount_base.html">13769</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/SwatchColorPicker-mount.html">13388</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/TagPicker-mount_base.html">3061</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/TagPicker-mount.html">3154</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/TeachingBubble-mount_base.html">102049</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/TeachingBubble-mount.html">115384</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Text-mount_base.html">506</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Text-mount.html">483</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/TextField-mount_base.html">1622</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/TextField-mount.html">1601</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ThemeProvider-mount_base.html">1351</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ThemeProvider-mount.html">1336</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ThemeProvider-virtual-rerender_base.html">733</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ThemeProvider-virtual-rerender.html">753</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1993</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2012</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Toggle-mount_base.html">907</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/Toggle-mount.html">931</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/buttonNative-mount_base.html">152</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24212/perf-test/buttonNative-mount.html">146</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: change/@fluentui-react-b6625e07-abc9-46fd-a63c-bc28f8aab0ef.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: add popupProps to Callout",
```
Comment (sopranopillow) [inline change/@fluentui-react-b6625e07-abc9-46fd-a63c-bc28f8aab0ef.json:3]: nit:
```suggestion
  "comment": "feat: Add popupProps to Callout.",
```

**Review by sopranopillow — APPROVED** [review]
lgtm!

File: packages/react/src/components/Callout/Callout.types.ts [hunk 2]
```diff
@@ -183,6 +184,11 @@ export interface ICalloutProps extends React.HTMLAttributes<HTMLDivElement>, Rea
    */
   layerProps?: ILayerProps;
 
+  /**
+   * Optional props to pass the Popup component that the panel uses.
```
Comment (smhigley) [inline packages/react/src/components/Callout/Callout.types.ts:188]:
```suggestion
   * Optional props to pass the Popup component that the callout uses.
```

**Review by spmonahan — APPROVED** [review]

**Pull request merged**